src/main/kotlin/com/autonomousapps/extension/SourceSetsHandler.kt

```diff
@@ -0,0 +1,61 @@
+package com.autonomousapps.extension
+
+import org.gradle.api.Named
+import org.gradle.api.NamedDomainObjectFactory
+import org.gradle.api.model.ObjectFactory
+import javax.inject.Inject
+
+/**
+ * Enable the user to specify custom filtering and severity behavior for a specific source set, either globally or for
+ * an individual project.
+ *
+ * This is not recommended. It would be much preferable for users to fix their dependency issues.
+ *
+ * ```
+ * dependencyAnalysis {
+ *   issues {
+ *     all {
+ *       sourceSet("...") {
+ *         onUnusedDependencies { ... }
+ *       }
+ *     }
+ *     project(":...") {
+ *       sourceSet("...") {
+ *         onUnusedDependencies { ... }
+ *       }
+ *     }
+ *   }
+ * }
+ * ```
+ */
+open class SourceSetsHandler @Inject constructor(
+  private val sourceSetName: String,
+  projectPath: String,
+  objects: ObjectFactory
+) : Named {
+
+  override fun getName(): String = sourceSetName
+
+  internal val project = objects.newInstance(ProjectIssueHandler::class.java, projectPath)
+
+  /**
+   * Gets the specified issue out of [project], and sets [sourceSetName] on it for later filtering by
+   * [FilterAdviceTask][com.autonomousapps.tasks.FilterAdviceTask].
+   */
+  internal fun issueOf(mapper: (ProjectIssueHandler) -> Issue): Issue {
+    return mapper(project).apply {
+      sourceSet.set(sourceSetName)
+    }
+  }
+
+  internal class Factory(
+    private val projectPath: String,
+    private val objects: ObjectFactory
+  ) : NamedDomainObjectFactory<SourceSetsHandler> {
+    override fun create(name: String): SourceSetsHandler = SourceSetsHandler(
+      sourceSetName = name,
+      projectPath = projectPath,
+      objects = objects
+    )
```

Comment:
```suggestion
    override fun create(name: String): SourceSetsHandler = objects.newInstance(
      SourceSetsHandler::class.java,
      name,
      projectPath
    )
```

Could do it like this, and have Gradle inject `objects` for SourceSetsHandler (which could then also be `abstract`)

Reply:
I'll give it a try. At the time of writing that code, I remembered being a little worried about the fact that both `name` and `projectPath` are Strings, Gradle would rely on the order of parameters which strikes me as fragile.